File: wwwroot/js/attendance.js
// Attendance page scripts
function toggleAttendance(checkbox, studentId) {
    const status = checkbox.checked;
    console.log(`Student ID ${studentId} attendance: ${status}`);
    // TODO: send AJAX to backend to mark attendance
}

function markAllPresent(courseId) {
    const checkboxes = document.querySelectorAll(".attendance-checkbox");
    checkboxes.forEach(cb => cb.checked = true);
    console.log(`All students marked present for course ${courseId}`);
}

File: Models/Attendance.cs
namespace CollegeManagementSystem.Models
{
    public class Attendance
    {
        public int Id { get; set; }
        public required string StudentId { get; set; }
        public required Student Student { get; set; }
        public int CourseId { get; set; }
        public required Course Course { get; set; }
        public DateTime Date { get; set; }
        public bool Status { get; set; } // true = present, false = absent
    }
}
